src/Controller/CRUDController.php

```diff
@@ -1290,27 +1217,19 @@ protected function validateCsrfToken($intention): void
 
     /**
      * Escape string for html output.
-     *
-     * @param string $s
-     *
-     * @return string
      */
-    protected function escapeHtml($s)
+    protected function escapeHtml(string $s): string
     {
         return htmlspecialchars((string) $s, ENT_QUOTES | ENT_SUBSTITUTE, 'UTF-8');
     }
 
     /**
      * Get CSRF token.
-     *
-     * @param string $intention
-     *
-     * @return string|false
      */
-    protected function getCsrfToken($intention)
+    protected function getCsrfToken(string $intention): ?string
     {
         if (!$this->has('security.csrf.token_manager')) {
-            return false;
+            return null;
```

Comment:
@VincentLanglet, shouldn't this change be exposed at `UPGRADE-4.0.md`?

Reply:
Oh yes indeed. Do you mind doing the PR ? I'm on phone currently.

Reply:
Yes, no problem. I'll do it later. Thank you!